org.civicrm.reqangular/test/services/api_test.js

```diff
@@ -16,107 +16,107 @@ define([
+      describe('when the API returns an error', function () {
         var returnValue = {
           is_error: 1,
           error_message: 'some error message'
         };
 
         beforeEach(function () {
-          $httpBackend.whenGET(/action\=send_get_action&entity\=send_get_entity/)
-            .respond(returnValue);
-
-          promise = api.sendGET(entity, action, params, cache);
+          promise = expectAndSendGET(returnValue)
         });
 
-        it('error message is thrown', function () {
+        it('rejects the promise with the error message provided by the API', function () {
           promise.catch(function (response) {
             expect(response).toBe(returnValue.error_message);
           });
         });
       });
+
+      function expectAndSendGET(returnValue) {
+        var action = 'get', entity = 'DummyEntity';
+
+        $httpBackend
+          .whenGET(new RegExp('action=' + action + '&entity=' + entity))
+          .respond(returnValue);
+
+        return api.sendGET(entity, action, {filter_name: 'filter_value'}, true);
+      }
     });
 
     describe('sendPOST', function () {
-      var entity = 'send_post_entity',
-        action = 'send_post_action',
-        params = {key: 'value'},
-        promise;
-
-      afterEach(function () {
-        $httpBackend.flush();
-      });
+      var promise;
 
-      describe('when there is no error', function () {
+      describe('when the API doesnt return an error', function () {
         var returnValue = {
           is_error: 0,
           somekey: 'someval'
         };
 
         beforeEach(function () {
-          $httpBackend.whenPOST('/civicrm/ajax/rest')
-            .respond(returnValue);
-
-          promise = api.sendPOST(entity, action, params);
+          promise = expectAndSendPOST(returnValue);
         });
 
-        it('proper value is returned', function () {
+        it('returns values sent from API', function () {
           promise.then(function (response) {
             expect(response).toEqual(returnValue);
           });
         });
       });
 
-      describe('when there is error', function () {
+      describe('when the API returns an error', function () {
         var returnValue = {
           is_error: 1,
           error_message: 'some error message'
         };
 
         beforeEach(function () {
-          $httpBackend.whenPOST('/civicrm/ajax/rest')
-            .respond(returnValue);
-
-          promise = api.sendPOST(entity, action, params);
+          promise = expectAndSendPOST(returnValue);
         });
 
-        it('error message is thrown', function () {
+        it('rejects the promise with the error message provided by the API', function () {
           promise.catch(function (response) {
             expect(response).toBe(returnValue.error_message);
           });
         });
       });
+
+      function expectAndSendPOST(returnValue) {
+        var entity = 'send_post_entity',
+          action = 'send_post_action';
```

Comment:
also, as already said you need to stick with the naming convention for both entities and actions in civicrm